src/Nethermind/Nethermind.Evm/EOF/EvmObjectFormat.cs

```diff
@@ -286,10 +288,90 @@ bool ValidateInstructions(ReadOnlySpan<byte> container, ref EofHeader? header)
                     header = null; return false;
                 }
 
+                if (_releaseSpec.StaticRelativeJumpsEnabled)
+                {
+                    if (opcode is Instruction.RJUMP or Instruction.RJUMPI)
+                    {
+                        if (i + 2 > sectionSize)
+                        {
+                            if (_loggingEnabled)
+                            {
+                                _logger.Trace($"EIP-4200 : Static Relative Jump Argument underflow");
+                            }
+                            header = null; return false;
+                        }
+
+                        var offset = code.Slice(i, 2).ReadEthInt16();
+                        immediates.Add(new Range(i, i + 1));
+                        var rjumpdest = offset + 2 + i;
+                        rjumpdests.Add(rjumpdest);
+                        if (rjumpdest < 0 || rjumpdest >= sectionSize)
+                        {
+                            if (_loggingEnabled)
+                            {
+                                _logger.Trace($"EIP-4200 : Static Relative Jump Destination outside of Code bounds");
+                            }
+                            header = null; return false;
+                        }
+                        i += 2;
+                    }
+
+                    if (opcode is Instruction.RJUMPV)
+                    {
+                        if (i + 2 > sectionSize)
```

Comment:
hmm the first if can be shared across all RJUMP opcodes

Reply:
Wouldn't extracting outside result in the addition of another if check, because this check is only done when opcode is either rjump, rjumpi or rjumpv
```
if(opcode is RJUMP or RJUMPI or RJUMPV) {
    if(index + 2 > size) {
        // YADAYADA
    } 
    if(opcode is RJUMP or RJUMPI) 
    {
        // Do Rjump and Rjumpi chexks
    } else {
        // Do Rjumpv checks
    } 
} 
```